zenodo/modules/records/serializers/schemas/common.py

```diff
@@ -459,6 +460,50 @@ def postload_references(self, data):
                 for ref in data['references'] if ref.strip()
             ]
 
+    custom = fields.Method('dump_custom', 'load_custom')
+
+    def load_custom(self, obj):
+        if not obj:
+            return missing
+        else:
+            custom_metadata_rules = \
+                current_app.config.get('ZENODO_CUSTOM_METADATA_DEFINITIONS')
+            for community in obj.keys():
+                if community not in custom_metadata_rules.keys():
+                    raise ValidationError(
+                        'The community: {0} is not supporting custom metadata'
+                        .format(community)
+                        )
+                for scheme in obj[community].keys():
```

Comment:
I guess we can rename `scheme` to `key`

```suggestion
                for key in obj[community]:
```